src/lib/utils/application-commands/computeDifferences.ts

```diff
@@ -0,0 +1,275 @@
+		const castedExisting = existingOption as APIApplicationCommandSubCommandOptions;
+		const castedExpected = option as APIApplicationCommandSubCommandOptions;
+
+		if (
+			castedExisting.type === ApplicationCommandOptionType.SubcommandGroup &&
+			castedExpected.type === ApplicationCommandOptionType.SubcommandGroup
+		) {
+			// We know we have options in this case, because they are both groups
+			for (const [subcommandIndex, subcommandOption] of castedExpected.options!.entries()) {
+				yield* reportOptionDifferences({
+					currentIndex: subcommandIndex,
+					option: subcommandOption,
+					existingOption: castedExisting.options?.[subcommandIndex],
+					keyPath: (index) => `${keyPath(currentIndex)}.options[${index}]`
+				});
+			}
+		} else if (
+			castedExisting.type === ApplicationCommandOptionType.Subcommand &&
+			castedExpected.type === ApplicationCommandOptionType.Subcommand
+		) {
+			// 0. No existing options and now we have options
+			if (!castedExisting.options?.length && castedExpected.options?.length) {
+				yield {
+					key: `${keyPath(currentIndex)}.options`,
+					expected: 'options present',
+					original: 'no options present'
+				};
+			}
+			// 1. Existing options and now we have no options
+			else if (castedExisting.options?.length && !castedExpected.options?.length) {
+				yield {
+					key: `${keyPath(currentIndex)}.options`,
+					expected: 'no options present',
+					original: 'options present'
+				};
+			}
+			// 2. Iterate over each option if we have any and see what's different
+			else if (castedExpected.options?.length) {
+				let processedIndex = 0;
+				for (const subcommandOption of castedExpected.options) {
+					const currentSubCommandOptionIndex = processedIndex++;
+					const existingSubcommandOption = castedExisting.options![currentSubCommandOptionIndex];
+
+					yield* reportOptionDifferences({
+						currentIndex: currentSubCommandOptionIndex,
+						option: subcommandOption,
+						existingOption: existingSubcommandOption,
+						keyPath: (index) => `${keyPath(currentIndex)}.options[${index}]`
+					});
+				}
+
+				// If we went through less options than we previously had, report that
+				if (processedIndex < castedExisting.options!.length) {
+					let option: APIApplicationCommandOption;
+					while ((option = castedExisting.options![processedIndex]) !== undefined) {
+						const expectedType =
+							optionTypeToPrettyName.get(option.type) ?? `unknown (${option.type}); please contact Sapphire developers about this!`;
+
+						yield {
+							key: `existing command option at path ${keyPath(currentIndex)}.options[${processedIndex}]`,
+							expected: 'no option present',
+							original: `${expectedType} with name ${option.name}`
+						};
+
+						processedIndex++;
+					}
+				}
+			}
+		}
+	}
+
+	// TODO(vladfrangu): publish discord-api-types already, so many new types to be had and validated >.> (min/max_value, autocomplete)
```

Comment:
Pretty sure you can remove this TODO now that DJS 13.5.0 is out.

Reply:
Removing the todo without implementing the checks? :P

Reply:
Then implement the checks 👀